eva-uow/src/main/kotlin/com/razz/eva/uow/UnitOfWorkExecutor.kt

```diff
@@ -63,66 +52,49 @@ class UnitOfWorkExecutor(
-        var uowSpan: Span? = null
+        lateinit var executorSpan: Span
         lateinit var name: String
         try {
             var currentAttempt = 0
             while (true) {
                 val uow = uowFactory()
                 if (currentAttempt == 0) {
                     name = uow.name()
-                    timer = createTimer(name)
-                    uowSpan = tracer.buildSpan(name).asChildOf(activeSpan).withTag(Tracing.Tags.UOW_NAME, name).start()
+                    executorSpan = executorSpan()
                 }
                 val constructedParams = params(InstantiationContext(currentAttempt))
-                val performSpan = buildPerformSpan(name, uowSpan)
-                val changes = withContext(ActiveSpanElement(performSpan) + PrimaryConnectionRequiredFlag) {
-                    try {
+                val changes = withContext(PrimaryConnectionRequiredFlag + executorSpan.asContextElement()) {
+                    uowSpan(name).use {
                         uow.tryPerform(principal, constructedParams)
-                    } catch (e: Exception) {
-                        performSpan.finishWithError(e)
-                        throw e
                     }
                 }
-                performSpan.finish()
-                val persistSpan = buildPersistSpan(name, uowSpan)
                 val persisted = try {
-                    persisting.persist(
-                        uowName = uow.name(),
-                        params = constructedParams,
-                        principal = principal,
-                        changes = changes.toPersist,
-                        clock = uow.clock(),
-                        uowSupportsOutOfOrderPersisting = uow.configuration().supportsOutOfOrderPersisting
-                    )
+                    withContext(executorSpan.asContextElement()) {
+                        persisting.persist(
+                            uowName = uow.name(),
+                            params = constructedParams,
+                            principal = principal,
+                            changes = changes.toPersist,
+                            clock = uow.clock(),
+                            uowSupportsOutOfOrderPersisting = uow.configuration().supportsOutOfOrderPersisting
+                        )
+                    }
                 } catch (e: PersistenceException) {
                     val config = uow.configuration()
-                    uowSpan?.log("persistence-exception")
                     if (config.retry.shouldRetry(currentAttempt, e)) {
                         currentAttempt += 1
                         logger.warn { "Retrying UnitOfWork: ${uow.name()}. Attempt: $currentAttempt" }
                         continue
                     }
-                    persistSpan.finishWithError(e)
                     return uow.onFailure(constructedParams, e)
-                } catch (e: Exception) {
-                    persistSpan.finishWithError(e)
```

Comment:
How to log errors in spans?

Reply:
I see it in helper [method](https://github.com/razz-team/eva/pull/142/files#diff-d3eec999099b05020867e8e64d54704311191072e526d5f15a076c5b2187ba85R17) but we are not calling this one here.